openapi-bal-task-plugin/src/main/java/io/ballerina/openapi/bal/tool/OpenAPICodeGeneratorTool.java

```diff
@@ -501,38 +506,62 @@ private static List<GenSrcFile> generateClientFiles(OASClientConfig oasClientCon
+                licenseContent + System.lineSeparator();
 
+        if (oasClientConfig.singleFile()) {
+            generateSingleFileForClient(toolContext, syntaxTree, ballerinaClientGenerator, sourceFiles, licenseHeader);
+        } else {
+            generateFilesForClient(syntaxTree, sourceFiles, licenseHeader, ballerinaClientGenerator);
+        }
+
+        return sourceFiles;
+    }
+
+    private static void generateFilesForClient(io.ballerina.compiler.syntax.tree.SyntaxTree syntaxTree,
+                                               List<GenSrcFile> sourceFiles, String licenseHeader,
+                                               BallerinaClientGenerator ballerinaClientGenerator) throws
+            FormatterException, IOException {
+        String mainContent = Formatter.format(syntaxTree).toSourceCode();
+        sourceFiles.add(new GenSrcFile(GenSrcFile.GenFileType.GEN_SRC, null, CLIENT_FILE_NAME,
+                licenseHeader + mainContent));
+        String utilContent = Formatter.format(
+                ballerinaClientGenerator.getBallerinaUtilGenerator().generateUtilSyntaxTree()).toString();
+        if (!utilContent.isBlank()) {
+            sourceFiles.add(new GenSrcFile(GenSrcFile.GenFileType.UTIL_SRC, null, UTIL_FILE_NAME,
+                    licenseHeader + utilContent));
+        }
+        // Generate ballerina records to represent schemas.
+        io.ballerina.compiler.syntax.tree.SyntaxTree schemaSyntaxTree = TypeHandler.getInstance()
+                .generateTypeSyntaxTree();
+        String schemaContent = Formatter.format(schemaSyntaxTree).toSourceCode();
         if (!schemaContent.isBlank()) {
             sourceFiles.add(new GenSrcFile(GenSrcFile.GenFileType.MODEL_SRC, null, TYPE_FILE_NAME,
-                    licenseContent == null || licenseContent.isBlank() ? schemaContent :
-                            licenseContent + System.lineSeparator() + schemaContent));
+                    licenseHeader + schemaContent));
         }
+    }
 
-        return sourceFiles;
+    private void generateSingleFileForClient(ToolContext toolContext,
+                                             io.ballerina.compiler.syntax.tree.SyntaxTree syntaxTree,
+                                             BallerinaClientGenerator ballerinaClientGenerator,
+                                             List<GenSrcFile> sourceFiles, String licenseHeader) throws IOException,
+            FormatterException {
+        syntaxTree = SingleFileGenerator.combineSyntaxTrees(syntaxTree,
+                ballerinaClientGenerator.getBallerinaUtilGenerator().generateUtilSyntaxTree(),
+                TypeHandler.getInstance().generateTypeSyntaxTree());
+        sourceFiles.add(new GenSrcFile(GenSrcFile.GenFileType.GEN_SRC, null,
+                CLIENT_FILE_NAME, licenseHeader + Formatter.format(syntaxTree).toSourceCode()));
+        deleteFileIfExists(toolContext.outputPath().resolve("types.bal"));
```

Comment:
What is the reason to delete this files here? Is this to remove the files generated in the previous run ?

Reply:
yes. if the ballerina.toml had `singleFile=false` in the previous run and now `singleFile=true`, the `types.bal` and `utils.bal` will still be there.